OpenSim/Framework/Communications/Cache/CachedUserInfo.cs

```diff
@@ -75,6 +75,15 @@ public class CachedUserInfo
         public bool HasReceivedInventory { get { return true; } }
 
         /// <summary>
+        /// This is a list of all of the inventory items that are of type "CallingCard" in the
+        ///   "Calling Card"/Friends/All folder to work around a bug with the viewer creating
+        ///   multiple copies of the same calling card
+        /// </summary>
+        private List<string> _namesOfCallingCardsInCallingCardsFriendsAllFolder = null;
+        private object _namesOfCallingCardsInCallingCardsFriendsAllFolderLock = new object();
```

Comment:
You would be correct in most cases, but you cannot lock on a null object - which _namesOfCallingCardsInCallingCardsFriendsAllFolder is initially. I could have made it so that it had a default value and then checked whether the count was 0, but I personally find that the null check makes it more clear that the cache hasn't been populated yet.